Spanner/src/Result.php

```diff
@@ -136,14 +141,16 @@ public function __construct(
         callable $call,
         $transactionContext,
         ValueMapper $mapper,
-        $retries = 3
+        $retries = 3,
+        &$transactionId = null
     ) {
         $this->operation = $operation;
         $this->session = $session;
         $this->call = $call;
         $this->transactionContext = $transactionContext;
         $this->mapper = $mapper;
         $this->retries = $retries;
+        $this->transactionId = &$transactionId;
```

Comment:
This type of behavior is strongly discouraged - this would mean that any changes to the passed in variable would update the variable here in the class, and vice-versa, e.g.:

```php
$transactionId = 123;
$result = new Result($transactionId);
$transactionId = 456;
var_dump($result->transactionId);  // int(456)
$result->transactionId = 789;
var_dump($transactionId);  // int(789)
```

This would lead to unexpected behavior in code, and is not something we want to support.

Reply:
Then we will require to send the Transaction object at least to make the communication between those classes possible.